Misc GeoPDF fixes

nyalldawson opened the pull request:
More GeoPDF related fixes

nyalldawson committed:
[geopdf] Expose option to set the initial visibility state of layers
in the created geopdf file from layouts

Reworks the layer structure section of the GeoPDF export options
dialog to add a new column allowing users to set the initial visibility
state of layers included in the PDF. This fixes a usability issue
with large generated GeoPDFs which can cause readers to grind to halt
when opening complex GeoPDF files with many layers.

Additionally, fixes an issue where users cannot set the logical
GeoPDF group for non-vector layers

Fixes #36536

nyalldawson committed:
[geopdf] Ensure that layer order in GeoPDF layer tree matches project
layer order

Fixes #34910

nyalldawson committed:
Remove unused variable

nyalldawson committed:
Respect layer opacity and composition modes where possible when exporting GeoPDF

Fixes #33305

nirvn commented:
GeoPDF format supports blending?! That's cool!

Is there a need to tweak the "warning, rasterization is suggested" advanced effect message here? I.e., if you are exporting to GeoPDF, supported layer blending modes shouldn't then throw a suggestion to rasterize the export.

nyalldawson committed:
Dox

nyalldawson committed:
Minor refactoring, more tests

nyalldawson committed:
Revise advanced effects warning to account for GeoPDF capabilities
when exporting canvas to GeoPDF

nyalldawson committed:
[layouts] Prevent GeoPDF exports when map items are rotated

(This is not supported in GDAL GeoPDF creation)

Fixes #34438

nyalldawson committed:
[layouts] Prevent atlas export to a single-file GeoPDF file

Instead show a warning the GeoPDF exports are only available when
the "export as single file" options is turned off.

Fixes #34024

nyalldawson commented:
@nirvn 
> Is there a need to tweak the "warning, rasterization is suggested" advanced effect message here? I.e., if you are exporting to GeoPDF, supported layer blending modes shouldn't then throw a suggestion to rasterize the export.

Done

nirvn commented:
Brilliant.
